### PR TITLE
Improve documentation and config validation of `loose` and `depth` parameters; drop support for `loose=None`

### PR DESCRIPTION
**Pull request by hoechenberger**
@larsoner @SophieHerbst WDYT?

We currently don't support volumetric or mixed source spaces anyway, so I thought we could simplify things a little.

I was even considering entirely removing the `loose` and `depth` settings but then wasn't sure if some of you might be using them sometimes? Personally I never touch those.

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: docs/source/v1.9.md.inc [hunk 1]
```diff
@@ -8,7 +8,13 @@
 
 [//]: # (- Whatever (#000 by @whoever))
 
-[//]: # (### :warning: Behavior changes)
+### :warning: Behavior changes
+
+- The [`loose`][mne_bids_pipeline._config.depth] parameter doesn't accept `"auto""`
```
Comment (hoechenberger) [inline docs/source/v1.9.md.inc:13]:
```suggestion
- The [`loose`][mne_bids_pipeline._config.loose] parameter doesn't accept `"auto""`
```

File: mne_bids_pipeline/_config.py [hunk 2]
```diff
@@ -1979,23 +1979,26 @@ def mri_landmarks_kind(bids_path):
...
-inverse solution is computed with **fixed orientation.**
-If `1`, it corresponds to **free orientation.**
-The default value, `'auto'`, is set to `0.2` for surface-oriented source
-spaces, and to `1.0` for volumetric, discrete, or mixed source spaces,
-unless `fixed is True` in which case the value 0. is used.
+A value that weights the source variances of the dipole components
+that are parallel (tangential) to the cortical surface.
+
+If `0`, then the inverse solution is computed with **fixed orientation**, i.e.,
+only dipoles perpendicular to the cortical surface are considered.
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:1988]:
```suggestion
only dipole components perpendicular to the cortical surface are considered.
```

File: mne_bids_pipeline/_config.py [hunk 3]
```diff
@@ -1979,23 +1979,26 @@ def mri_landmarks_kind(bids_path):
...
+
+If `1`, it corresponds to **free orientation**, i.e., dipoles with any orientation are
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:1990]:
```suggestion
If `1`, it corresponds to **free orientation**, i.e., dipole components with any orientation are
```

**Comment by larsoner** [comment]
Eventually we could pretty easily add support for volumetric source estimates I think. So I'm not sure it's worth getting rid of `'auto'` just to potentially add it back. And I think `loose` and `depth` are both useful from time to time, especially when working with template / non-individualized MRIs (you often want `loose=1.`)

**Comment by SophieHerbst** [comment]
I agree about keeping the loose and depth parameters, we do use them sometimes. 
No opinion about the 'auto' option, but if mixed source spaces can be implemented easily, that would be useful.

**Comment by SophieHerbst** [comment]
I have always found the options for those parameters a bit confusing, so I am fine with removing the None.

**Comment by larsoner** [comment]
Okay with me to remove `None`

File: mne_bids_pipeline/_config.py [hunk 4]
```diff
@@ -1979,24 +1979,33 @@ def mri_landmarks_kind(bids_path):
 
 # ## Inverse solution
 
-loose: float | Literal["auto"] = 0.2
+loose: Annotated[float, Interval(ge=0, le=1)] = 0.2
```
Comment (larsoner) [inline mne_bids_pipeline/_config.py:1982]: To restore old `auto` behavior, something like
```suggestion
loose: Annotated[float, Interval(ge=0, le=1)] | Literal["auto"] = 0.2
```

Comment (hoechenberger) [reply mne_bids_pipeline/_config.py]: right, forgot about that one!!

**Review by larsoner — APPROVED** [review]
Marking for merge when green, thanks in advance @SophieHerbst !

**Comment by larsoner** [comment]
Whoops wrong PR comment, thanks @hoechenberger !

**Pull request merged**